libmcount/plthook.c

```diff
@@ -923,6 +919,10 @@ unsigned long plthook_exit(long *retval)
 	ret_loc  = rstack->parent_loc;
 	ret_addr = rstack->parent_ip;
 
+	pr_dbg3("[idx: %4d] exit  %lx: %s     (resolved addr: %lx)\n",
```

Comment:
If you think so, what about this?
```diff
diff --git a/libmcount/plthook.c b/libmcount/plthook.c
index 8d067bf2..caf9fbb2 100644
--- a/libmcount/plthook.c
+++ b/libmcount/plthook.c
@@ -927,8 +927,8 @@ again:
        ret_loc  = rstack->parent_loc;
        ret_addr = rstack->parent_ip;
 
-       pr_dbg3("[idx: %4d] exit  %lx: %s     (resolved addr: %lx)\n",
-               dyn_idx, ret_addr, rstack->pd->dsymtab.sym[dyn_idx].name,
+       pr_dbg3("[idx: %4d] exit  %lx: %s     (return addr: %lx)\n",
+               dyn_idx, rstack->pd->dsymtab.sym[dyn_idx].name, ret_addr,
                rstack->pd->resolved_addr[dyn_idx]);
 
        /* re-hijack return address of parent */
```
I would like to see what `plthook_exit` returns.

Reply:
> Also, to match with the entry line, wouldn't it be better adding @plt suffix here as well?

I didn't add `@plt` on purpose because `plthook_exit` is entered from the actual library function not from `@plt` function. It's to distinguish between them although it might be just a matter of preference. If you disagree, then I will add `@plt` here as well.

Reply:
Hmm.. thinking it again. The output log doesn't look straight forward to me if resolved address comes first.
```
  plthook: [idx: 2] enter 4005b0: __monstartup@plt   (mod: 7f37a39e6168)
  plthook: [idx: 2] exit  7f37a3590bf0: __monstartup (return addr: 400671)
  plthook: [idx: 4] enter 4005d0: __cxa_atexit@plt   (mod: 7f37a39e6168)
  plthook: [idx: 4] exit  7f37a31f7280: __cxa_atexit (resolved addr: 400575)
```
I think the current change looks better than this.
```
  plthook: [idx: 2] enter 4005b0: __monstartup@plt (mod: 7f37a39e6168)
  plthook: [idx: 2] exit  400671: __monstartup     (resolved addr: 7f37a3590bf0)
  plthook: [idx: 4] enter 4005d0: __cxa_atexit@plt (mod: 7f37a39e6168)
  plthook: [idx: 4] exit  400575: __cxa_atexit     (resolved addr: 7f37a31f7280)
```

Reply:
I feel more like “enter” for function entry and “exit” for function return (rather than resolved address).

Reply:
I think it's a matter of how we interpret the address in the message.  But yes, your way is also reasonable so it's good to leave as is.

Reply:
Thanks. I just updated without changing this.